src/classes/RD2_ValidationService_TEST.cls

```diff
@@ -0,0 +1,196 @@
+        selected HouseHold account. It also verifies if no error message when the contact is associated
+        with the account in this method.
+    */
+    @isTest
+    private static void shouldFailWhenContactDoesNotBelongToHHAccount() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Account acc = getHHAccount();
+        insert acc;
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withAccount(acc.Id)
+            .withContact(con.Id)
+            .build();
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        System.assert(errMessage.contains(System.Label.RD_ContactMustBelongToAccount));
+    }
+
+    /**
+    * @description Should fail when there is an invalid contact for the account in the
+        Recurring Donations list.
+    */
+    @isTest
+    private static void shouldFailOnlyInvalidRecurringDonations() {
+
+        String errMessage ='';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        List<Contact> contacts = new List<Contact>{
+            UTIL_UnitTestData_TEST.getContact(),
+            UTIL_UnitTestData_TEST.getContact()
+        };
+        insert contacts;
+
+        contacts = [SELECT AccountId FROM Contact WHERE Id IN :contacts];
+
+        List<npe03__Recurring_Donation__c> rds = new TEST_RecurringDonationBuilder()
+        .withAccount(contacts[0].AccountId)//only RD for one Contact will be valid
+        .withAmount(100)
+        .build(contacts);
+
+        List<Database.SaveResult> results = Database.insert(rds, false);
+        ERR_Handler.Errors errors = ERR_Handler.getErrors(results, rds);
+        System.assertEquals(true, errors.errorsExist);
+        System.assertEquals(1, errors.errorRecords.size());
+
+    }
+
+    /***
+    * @description Verifies if it creates Recurring donations for a single household contact
+        without an account.
+    */
+    @isTest
+    private static void shouldCreateRDSForHouseholdContact() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .build();
+
+        insert rd;
+
+        System.assert([select id from npe03__Recurring_Donation__c].size()>0 );
+
+    }
+
+    /***
+    * @description Should create Recurring donations for organization accounts without validaiton errors
+    */
+    @isTest
+    private static void shouldCreateRDSForOrganizationAccounts() {
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+        Account account = new Account(Name = 'TEST_ACCOUNT');
```

Comment:
Let's utilize the test data factory to build the Account, ie 
```
Account account = UTIL_UnitTestData_TEST.createMultipleTestAccounts(1, CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE)[0];
```

Reply:
> Let's utilize the test data factory to build the Organization Account, ie
> 
> ```
> Account account = UTIL_UnitTestData_TEST.createMultipleTestAccounts(1, CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE)[0];
> ```

@RadSFDeveloper , one to one organization type is one to one individual account and not organization account, therefore the test is failing because of that. I am going to change back the code.